source/common/upstream/health_discovery_service.cc

```diff
@@ -189,6 +192,18 @@ void HdsDelegate::onReceiveMessage(
   stats_.requests_.inc();
   ENVOY_LOG(debug, "New health check response message {} ", message->DebugString());
 
+  // Validate message fields
+  try {
+    MessageUtil::validate(*message, validation_visitor_);
+  } catch (const ProtoValidationException& ex) {
+    // Increment error count
+    stats_.errors_.inc();
+    ENVOY_LOG(warn, "unable to validate hds proto: {}", ex.what());
```

Comment:
Nit: `"Unable to validate health check specifier: {}"`

Reply:
I agree that this is more clear, code now reflects this.